### PR TITLE
Progress Bar created

### LINKED ISSUE #20: Use ProgressBar
ProgressDialog has been deprecated

### PR DESCRIPTION
**Pull request by dreadpool2**
Fixes #20
videotogif_2018 05 29_21 42 52

### CODE REVIEW

**Comment by simarsingh24** [comment]
@dreadpool2 always attach a gif/screenshot in these kinds of PR's

**Comment by dreadpool2** [comment]
@iamareebjamal Why is the build failing? Its working fine here

**Comment by dreadpool2** [comment]
In the details it states that it is an error due to proguard. Should I disable it?

File: app/build.gradle [hunk 1]
```diff
@@ -61,6 +61,7 @@ dependencies {
 
     //picasso
     implementation 'com.squareup.picasso:picasso:2.71828'
+    implementation 'com.jakewharton.picasso:picasso2-okhttp3-downloader:1.1.0'
```
Comment (iamareebjamal) [inline app/build.gradle]: Why is this needed?

**Comment by dreadpool2** [comment]
videotogif_2018 05 29_23 48 35

**Comment by dreadpool2** [comment]
@iamareebjamal done with the change

File: app/src/main/res/layout/activity_login.xml [hunk 2]
```diff
@@ -50,7 +50,7 @@
         </android.support.design.widget.TextInputLayout>
 
         <android.support.v7.widget.AppCompatButton
-            android:id="@+id/login"
+            android:id="@+id/login_button"
```
Comment (iamareebjamal) [inline app/src/main/res/layout/activity_login.xml]: Camel case

File: app/src/main/java/org/fossasia/openevent/general/LoginActivity.kt [hunk 3]
```diff
@@ -54,13 +51,15 @@ class LoginActivity : AppCompatActivity() {
...
+                    login_button.isEnabled = true
+                    progressBar.visibility = View.GONE
                     SharedPreferencesUtil.putString(ConstantStrings.TOKEN, accessToken)
                     if (accessToken != null)
                         redirectToMain()
                 }) { throwable ->
                     ApiClient.setToken(null)
-                    progressDialog.cancel()
+                    login_button.isEnabled = true
+                    progressBar.visibility = View.GONE
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/LoginActivity.kt]: Move these two lines in a function with boolean parameter

**Comment by dreadpool2** [comment]
@iamareebjamal Did the changes

File: app/src/main/java/org/fossasia/openevent/general/LoginActivity.kt [hunk 4]
```diff
@@ -70,4 +66,12 @@ class LoginActivity : AppCompatActivity() {
         super.onDestroy()
         compositeDisposable.dispose()
     }
+
+    private fun loginEnabled(boolean:Boolean) {
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/LoginActivity.kt]: enabled: Boolean

File: app/src/main/java/org/fossasia/openevent/general/LoginActivity.kt [hunk 5]
```diff
@@ -70,4 +66,12 @@ class LoginActivity : AppCompatActivity() {
...
+        loginButton.isEnabled = boolean
+        progressBar.visibility = View.VISIBLE
+        if(boolean) {
+            progressBar.visibility = View.GONE
+        }
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/LoginActivity.kt]: progressBar.visibility = if enabled View.GONE else View.VISIBLE

File: app/src/main/res/layout/activity_login.xml [hunk 6]
```diff
@@ -67,5 +67,12 @@
             android:text="@string/sign_in"
             android:textSize="@dimen/text_size_large" />
 
+        <ProgressBar
+            android:layout_width="wrap_content"
+            android:layout_gravity="center"
+            android:id="@+id/progressBar"
+            android:visibility="gone"
+            android:layout_height="wrap_content" />
```
Comment (iamareebjamal) [inline app/src/main/res/layout/activity_login.xml]: Height and width should be together

File: app/src/main/java/org/fossasia/openevent/general/LoginActivity.kt (hunk 4 above)
Comment (iamareebjamal) [reply app/src/main/java/org/fossasia/openevent/general/LoginActivity.kt]: Rename to showProgress

Naming of functions is very important and needs to reflect their major task

**Comment by dreadpool2** [comment]
Okay :+1:

**Comment by dreadpool2** [comment]
Are the changes fine now?

File: app/src/main/java/org/fossasia/openevent/general/LoginActivity.kt [hunk 7]
```diff
@@ -70,4 +66,9 @@ class LoginActivity : AppCompatActivity() {
         super.onDestroy()
         compositeDisposable.dispose()
     }
+
+    private fun showProgress(enabled:Boolean) {
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/LoginActivity.kt]: Read kotlin style guideline, there should be space after : 

I'm accepting for now. Good job

**Comment by iamareebjamal** [comment]
@dreadpool2 Also please take a thought when writing commit messages. They are really important https://chris.beams.io/posts/git-commit/

**Comment by dreadpool2** [comment]
@iamareebjamal Sure :+1:
